lib/galaxy/workflow/run.py

```diff
@@ -165,8 +166,9 @@ def invoke( self ):
                     # https://github.com/galaxyproject/galaxy/issues/2259
                     if job:
                         workflow_invocation_step.job_id = job.id
-            except modules.DelayedWorkflowEvaluation:
+            except modules.DelayedWorkflowEvaluation as de:
                 step_delayed = delayed_steps = True
+                more_info = "(%s)" % de.why
                 self.progress.mark_step_outputs_delayed( step )
```

Comment:
It isn't quite how I think about this - but you are right the code is more consistent and the logs look better - so it makes me think I'm thinking about this wrong. Give me a second and I'll rebase.